packages/react/src/components/library/change-profile-password-form/ChangeProfilePasswordForm.tsx

```diff
@@ -0,0 +1,130 @@
+import './ChangeProfilePasswordForm.scss';
+import React, { useState, useCallback, useMemo, useRef, useEffect } from 'react';
+import notify from 'devextreme/ui/notify';
+import { ValidationRule } from 'devextreme/ui/validation_rules';
+import Form, { Item, Label } from 'devextreme-react/form';
+import Validator from 'devextreme-react/validator';
+import { FormPopup } from '../../utils/form-popup/FormPopup';
+import { PasswordTextBox } from '../password-text-box/PasswordTextBox';
+import { ValidatedEvent } from 'devextreme/ui/validator';
+
+const saveNewPassword = (): void => {
+  notify({ message: 'Password Changed', position: { at: 'bottom center', my: 'bottom center' } }, 'success');
+};
+
+export const ChangeProfilePasswordForm = ({ visible, setVisible }) => {
+  const confirmField = useRef<Validator>(null);
+  const [currentPassword, setCurrentPassword] = useState('');
+  const [newPassword, setNewPassword] = useState('');
+  const [confirmedPassword, setConfirmedPassword] = useState('');
+  const [currentPasswordValid, setCurrentPasswordValid] = useState(false);
+  const [newPasswordValid, setNewPasswordValid] = useState(false);
+  const [confirmedPasswordValid, setConfirmedPasswordValid] = useState(false);
+  const [isSaveDisabled, setIsSaveDisabled] = useState(true);
+
+  const confirmPasswordValidators = useMemo((): ValidationRule[] => {
+    return [{
+      type: 'compare',
+      message: 'Passwords do not match',
+      comparisonTarget: () => newPassword,
+    }];
+  }, [newPassword]);
+
+  useEffect(() => {
+    const formValues = [currentPassword, newPassword, confirmedPassword];
+    const validity = [currentPasswordValid, newPasswordValid, confirmedPasswordValid];
+
+    setIsSaveDisabled(
+      formValues.some((value) => !value) ||
+      validity.some((value) => !value)
+    );
+  }, [
+    currentPassword,
+    newPassword,
+    confirmedPassword,
+    currentPasswordValid,
+    newPasswordValid,
+    confirmedPasswordValid
+  ]);
+
+  const checkConfirm = useCallback(() => {
+    confirmField.current?.instance.validate();
+  }, []);
+
+  const onCurrentPasswordValidated = useCallback((e: ValidatedEvent) => {
+    setCurrentPasswordValid(!!e.isValid);
+  }, []);
+
+  const onCurrentPasswordChange = useCallback((value) => {
+    setCurrentPassword(value);
+  }, []);
+
+  const onConfirmedPasswordValidated = useCallback((e: ValidatedEvent) => {
+    setConfirmedPasswordValid(!!e.isValid);
+  }, []);
+
+  const onConfirmPasswordChange = useCallback((value) => {
```

Comment:
From my point of view, this callback looks excessive and you can just use setConfirmedPassword as onValueChange prop value